draft-ietf-tls-esni.md

```diff
@@ -1252,16 +1238,83 @@ TCP connections an attacker can open.
+the protocol and measuring its impact. [[OPEN ISSUE: Run an experiment that
+compares the error rate of GREASE ECH connections with vanilla TLS 1.3.]]
+
+From this experiment we can deduce the set of features of the TLS handshake that
+are known to trigger differentiable treatment by the network. This set might
+include the plaintext extensions offered by the client or server, plaintext
+alert messages, the number of round trips over the network, the length of each
+message, and so on. We will call these the protocol's "observable features". Our
+goal is to design real ECH so that each execution of the protocol has the same
+observable features as if GREASE ECH were being executed instead. For example,
+if extensions are an observable feature, then regardless of whether they are
+running the target protocol or the cover protocol, the client and server should
+appear to negotiate the same set of extensions. That way the real protocol does
+not trigger differentiable treatment that would not have been triggered by the
+cover protocol.
+
+We say that the target protocol is "indifferentiable" from cover protocol if,
+when executed in the presence of the adversary, both protocols have the same set
+of observable features. The strength of this model depends on how rich the set
+of observable features is: it may include the transcript of the protocol's
+execution itself, yielding a very strong attacker; or it may be as coarse
+grained as the number of IP packets sent. The model's strength also depends on
+whether the attacker is passive or active (see {{goals}}); middleboxes have been
+known to exhibit active behavior, even without malicious intent [[TOOD: Add
+references]].
+
+Indifferentiability of ECH from GREASE ECH in a strong attack model is not
+feasible for all implementations. Thus, this specification aims to provide a
+baseline security level that most deployments can achieve easily, while
+providing implementations enough flexibility to achieve stronger security where
+possible. Minimally, real ECH is designed to be indifferentiable from GREASE ECH
+for passive adversaries with following capabilities:
+1. The attacker does not know the ECHConfigs used by the server.
+1. The attacker keeps per-connection state only. In particular, it does not
+   track endpoints across connections.
+1. Observable features include: the code points of extensions negotiated in the
+   clear, but not the payloads themselves; the length of messages; and the
+   values of plaintext alert messages.
+
+This leaves a variety of practical differentiators out-of-scope. including,
+though not limited to, the following:
+1. the value of the configuration identifier;
+1. the value of the outer SNI;
+1. use of the "pre_shared_key" extension in the ClientHelloOuter, which is
+   permitted in GREASE ECH but not real ECH; [[TODO: Based on discussion in
+   #354, we intend to mitigate this differentiator.]]
```

Comment:
This isn't clear to me. I'd just leave this as an open issue without a decision one way or the other.

```suggestion
   #<new issue ID>, this may be mitigated with a future change.]]
```